components/InputTypeDropzone.js

```diff
@@ -1,98 +1,82 @@
-import React from 'react';
+import React, { useState, useCallback } from 'react';
 import PropTypes from 'prop-types';
-import Dropzone from 'react-dropzone';
+import { useDropzone } from 'react-dropzone';
 import { get } from 'lodash';
 import { defineMessages, injectIntl } from 'react-intl';
 import { themeGet } from '@styled-system/theme-get';
 
 import { imagePreview } from '../lib/utils';
 import { upload } from '../lib/api';
 
-class InputTypeDropzone extends React.Component {
-  static propTypes = {
-    defaultValue: PropTypes.string,
-    className: PropTypes.string,
-    onChange: PropTypes.func.isRequired,
-    options: PropTypes.object,
-    intl: PropTypes.object.isRequired,
-    placeholder: PropTypes.string,
-    name: PropTypes.string,
-  };
+const InputTypeDropzone = props => {
+  const { defaultValue, className, onChange, options, intl, placeholder, name } = props;
+  const [loading, setLoading] = useState(false);
+  const [value, setValue] = useState(defaultValue);
+  const [url, setUrl] = useState(defaultValue);
+  const [error, setError] = useState(null);
 
-  constructor(props) {
-    super(props);
-    this.renderContainer = this.renderContainer.bind(this);
-    this.handleChange = this.handleChange.bind(this);
-    this.state = {
-      loading: false,
-      value: props.defaultValue,
-      url: props.defaultValue,
-    }; // value can be base64 encoded after upload, url is always an url
-    this.messages = defineMessages({
-      placeholder: {
-        id: 'uploadImage.placeholder',
-        defaultMessage: 'Drop an image or click to upload',
-      },
-      isDragActive: {
-        id: 'uploadImage.isDragActive',
-        defaultMessage: "Drop it like it's hot 🔥",
-      },
-      isDragReject: {
-        id: 'uploadImage.isDragReject',
-        defaultMessage: '🚫 This file type is not accepted',
-      },
-      error: { id: 'uploadImage.error', defaultMessage: 'Error: {error}' },
-    });
-  }
+  const messages = defineMessages({
```

Comment:
This is calling `defineMessages` at each render. Can we move it outside of the function?